<launch>
  <arg name="steer_update_hz" default="10.0"/>
  <arg name="initial_covariance" default="1000.0"/>
  <arg name="forgetting_factor" default="0.999"/>
  <arg name="valid_min_velocity" default="5.0"/>
  <arg name="valid_max_steer" default="0.05"/>
  <!-- ndt twist publisher -->
  <include file="$(find-pkg-share pose2twist)/launch/pose2twist.launch.xml">
    <arg name="input_pose_topic" value="/localization/pose_estimator/pose"/>
    <arg name="output_twist_topic" value="/ndt_twist"/>
  </include>
  <!-- steer offset estimator -->
  <node pkg="steering_offset_estimator" exec="steering_offset_estimator" name="steering_offset_estimator" output="screen">
    <arg name="steer_update_hz" value="$(var steer_update_hz)"/>
    <arg name="initial_covariance" value="$(var initial_covariance)"/>
    <arg name="forgetting_factor" value="$(var forgetting_factor)"/>
    <arg name="valid_min_velocity" value="$(var valid_min_velocity)"/>
    <arg name="valid_max_steer" value="$(var valid_max_steer)"/>
    <remap from="input/twist" to="/ndt_twist"/>
    <remap from="input/steer" to="/vehicle/status/steering"/>
    <remap from="output/steering_offset" to="/vehicle/status/steering_offset" />
    <remap from="output/steering_offset_covariance" to="/vehicle/status/steering_offset_covariance" />
  </node>
</launch>
